pkg/location/reference.go

```diff
@@ -0,0 +1,13 @@
+package location
+
+import "fmt"
+
+type Reference interface {
+	fmt.Stringer
+	Type() string
+	Validate() error
+}
+
+type ReferenceLock interface {
+	Reference
```

Comment:
Since we have SetLock interface, I wonder if a counterpart here would be:

`Lock() string`

it may be nice to glean some useful information from the interface